internal/pkg/plugin/go_plugin_installer.go

```diff
@@ -25,9 +29,8 @@ func (g *GoPluginInstaller) CheckVersion(ver *version.Version) error {
 		return errors.NewErrorWithSuggestions(fmt.Sprintf(programNotFoundErrorMsg, "go"), programNotFoundSuggestions)
 	}
 
-	re := regexp.MustCompile(goVersionPattern)
 	for _, word := range strings.Split(string(out), " ") {
-		if re.MatchString(word) {
+		if g.IsVersionNumber(word) {
 			installedVer, err := version.NewVersion(strings.TrimPrefix(word, "go"))
 			if err != nil {
 				return errors.Errorf(unableToParseVersionErrorMsg, "go")
```

Comment:
I think we may have discussed this earlier, but I wonder if this should actually be a warning. If the user ends up having a old version of a dependency that happens to still work, it would be nice if the plugin still ran.